src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -657,6 +649,9 @@ private static IEnumerable<TSource> TakeWhileIterator<TSource>(IEnumerable<TSour
         public static IEnumerable<TSource> Skip<TSource>(this IEnumerable<TSource> source, int count)
         {
             if (source == null) throw Error.ArgumentNull("source");
+            if (count < 0) count = 0;
```

Comment:
"return source" might not be a good idea because someone might use Skip(0) as a way to wrap an otherwise mutable source (like an array) into an iterator. I actually have seen code like that.

The idea is to make it hard to get to the underlying data and change it. It might still be possible through some reflection/unsafe code, but that woudl be considerably harder and clearly a dependency on underlying implementation compared to just casting result to T[].